main.go

```diff
@@ -2,166 +2,77 @@ package main
 	"time"
 
-	"github.com/elastic/hey-apm/cli"
-	"github.com/elastic/hey-apm/out"
-	"github.com/elastic/hey-apm/requester"
-	"github.com/elastic/hey-apm/target"
+	"go.elastic.co/apm"
+	apmtransport "go.elastic.co/apm/transport"
 )
 
-var (
+func main() {
 	// run options
-	runTimeout  = flag.Duration("run", 30*time.Second, "stop run after this duration")
-	maxRequests = flag.Int("requests", math.MaxInt32, "maximum requests to make")
-
-	// payload options
-	numAgents       = flag.Int("c", 1, "number of agents sending data concurrently")
-	qps             = flag.Float64("q", 0, "queries per second")
-	pause           = flag.Duration("p", 1*time.Millisecond, "Only used if `qps` is not set. Defines the pause between sending events over the same http request.")
-	numErrors       = flag.Int("e", 3, "number of errors")
-	numTransactions = flag.Int("t", 6, "number of transactions")
-	numFrames       = flag.Int("f", 20, "number of stacktrace frames per span")
-	numSpans        = flag.Int("s", 7, "number of spans")
+	runTimeout := flag.Duration("run", 30*time.Second, "stop run after this duration")
 
-	// http options
-	baseUrl            = flag.String("base-url", "http://localhost:8200", "")
-	endpoint           = flag.String("endpoint", "/intake/v2/events", "")
-	method             = flag.String("method", "POST", "method type")
-	headers            = newStringsOpt("header", []string{}, "header(s) added to all requests")
-	requestTimeout     = flag.Duration("request-timeout", 10*time.Second, "request timeout in seconds")
-	idleTimeout        = flag.Duration("idle-timeout", 3*time.Minute, "idle timeout")
-	disableCompression = flag.Bool("disable-compression", false, "")
-	disableKeepAlives  = flag.Bool("disable-keepalive", false, "")
-	disableRedirects   = flag.Bool("disable-redirects", false, "")
-
-	interactive = flag.Bool("cli", false, "run hey-apm in interactive mode listening on 8234")
-	stream      = flag.Bool("stream", false, "send data in a streaming way via http")
-)
-
-type stringsOpt struct {
-	val []string
-	set bool
-}
-
-func (s *stringsOpt) String() string {
-	return fmt.Sprint(*s)
-}
-
-func (s *stringsOpt) Set(value string) error {
-	if !s.set {
-		(*s).val = []string{}
-	}
-	s.set = true
-	(*s).val = append((*s).val, value)
-	return nil
-}
-
-func (s *stringsOpt) List() []string {
-	return []string((*s).val)
-}
-
-func newStringsOpt(name string, value []string, usage string) *stringsOpt {
-	var s stringsOpt
-	for _, v := range value {
-		s.val = append(s.val, v)
-	}
-	flag.Var(&s, name, usage)
-	return &s
-}
+	// apm-server options
+	apmServerSecret := flag.String("secret", "", "") // ELASTIC_APM_SECRET_TOKEN
+	apmServerUrl := flag.String("url", "", "")       // ELASTIC_APM_SERVER_URL
 
-func main() {
+	// payload options
+	//errorLimit := flag.Int("e", math.MaxInt64, "max errors to generate")
+	spanMaxLimit := flag.Int("sx", 10, "max spans to per transaction")
+	spanMinLimit := flag.Int("sm", 1, "min spans to per transaction")
```

Comment:
not a big deal, but i'd set the same default for `sm` and `sx` to minimize randomness (in a default run, that is)

Reply:
Skipping this for now.  I think it will be nice to be able to hand this binary to a tester and have it generate unique looking traces.  For benchmarking we'll be sure to set all options explicitly, and of course capture them in the report.